subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -860,6 +877,10 @@ private boolean noDaemonArgumentGiven() {
             allArgs.add("--console=" + consoleType.toString().toLowerCase());
         }
 
+        for (WarningType type : warningTypes) {
+            allArgs.add("--warnings=" + type.getBuildOption());
```

Comment:
:x: What exactly happens if you use multiple warning types? What's the use case? Shouldn't there be just one?

Reply:
In the long term we may allow `--warnings=deprecation,compiler,<category>` etc. Please see discussion above.